From cdddc24c05cedc0d0fabc48da17ec26aeb7c1ef9 Mon Sep 17 00:00:00 2001
From: Tobias Waldekranz <tobias@waldekranz.com>
Date: Wed, 27 Mar 2024 10:10:19 +0100
Subject: [PATCH 22/34] net: phy: Do not resume PHY when attaching
Organization: Addiva Elektronik

The PHY should not start negotiating with its link-partner until
explicitly instructed to do so.

Therefore, skip resuming the PHY in the attach phase, deferring it to
when phy_start() is called from the net_device's open() callback,
possibly via phylink_start().

Otherwise, drivers that attached to their PHYs during
probing (e.g. DSA) would end up with a physical link being
established, even though the corresponding interface was still
administratively down.

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 drivers/net/phy/phy_device.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index 1b144e23911c..16c9938465a2 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -1555,7 +1555,6 @@ int phy_attach_direct(struct net_device *dev, struct phy_device *phydev,
 	if (err)
 		goto error;
 
-	phy_resume(phydev);
 	if (!phydev->is_on_sfp_module)
 		phy_led_triggers_register(phydev);
 
-- 
2.43.0

